Preference page

cknguyen123 opened the pull request.

cknguyen123 committed:
#64 - UI and Preference Activity Implementation
Created a simple UI for preference, Created Activity class for Preference to begin linking actions

cknguyen123 committed:
#27 - gettings and setting data from/to firebase
Made other UI fixes and loading from database

LoukikNaik commented:
The bio editText field gets hidden behind the keyboard when the keyboard pops up.
Wrapping constrained layout in `fragment_preferences.xml` with `ScrollView` seems to solve the problem.

LoukikNaik commented:
`<ScrollView
    xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:fillViewport="true"
    android:layout_width="match_parent"
    android:layout_height="match_parent">`

cknguyen123 committed:
fix for keyboard covering bio

cknguyen123 committed:
#69 Updating UI based off db info
-Made sure information is populated on preference start up.
-Made sure all fields were inputted before allowing to save
-Made activity and location fields

cknguyen123 committed:
Cleaning up onCreate by organizing actions into functions

cknguyen123 committed:
Cleaning up onCreate by organizing actions into functions

cknguyen123 committed:
Reorganizing functions for unit testing

cknguyen123 committed:
UI testing for preference page

cknguyen123 committed:
UI testing for preference page

cknguyen123 committed:
updated version of PreferenceActivity

cknguyen123 committed:
updated version of PreferenceActivity

amquach00 commented on app/src/main/java/com/example/activeamigo/PreferenceActivity.java:
```diff
@@ -0,0 +1,333 @@
+    protected String gender;
+    protected EditText bio;
+
+    protected String collection="Accounts";
```
please make static

cknguyen123 replied:
needs to be private and modifiable. cannot be static

amquach00 commented on app/src/main/java/com/example/activeamigo/PreferenceActivity.java:
```diff
@@ -0,0 +1,333 @@
+    protected EditText bio;
+
+    protected String collection="Accounts";
+    protected String document = "test@ucsd.edu";
```
please make static

amquach00 commented on app/src/main/java/com/example/activeamigo/PreferenceActivity.java:
```diff
@@ -0,0 +1,333 @@
+
+public class PreferenceActivity extends AppCompatActivity {
+
+    protected EditText dobEditText;
```
please make variables on line 37-43 private

amquach00 commented on app/src/main/java/com/example/activeamigo/PreferenceActivity.java:
```diff
@@ -0,0 +1,333 @@
+    private boolean checkFieldsCompleted(){
+        if (exercise_choice.getSelectedItemPosition() == 0 || location_choice.getSelectedItemPosition() == 0) {
+            // Spinner is at index 0 (first item) indicating "Select Activity" or "Select Location" is still selected
+            Toast.makeText(this, "Please select valid options for activity and location", Toast.LENGTH_SHORT).show();
```
add these strings in the strings.xml file

amquach00 reviewed (dismissed):
everything lgtm

cknguyen123 committed:
Code review changes

cknguyen123 committed:
Merge branch 'main' of https://github.com/CSE-210/Exercise-Buddy-App into PreferencePage

cknguyen123 committed:
#19 Ready to merge to complete user story #3

cknguyen123 committed:
#19 Ready to merge to complete user story,

cknguyen123 committed:
#19 Ready to merge to complete user story,
